### PR TITLE
Explore: updates explore table container to show a span on 0 series returned

### LINKED ISSUE #21476: Explore view doesn't show anything when you do a bad query
Might be worth saying "0 series returned" or something:

Screenshot 2020-01-13 at 13 04 05

### PR DESCRIPTION
**Pull request by Estrax**
**What this PR does / why we need it**:

Updates explore table container to display a span when no series are returned.

Screenshot

**Which issue(s) this PR fixes**:

Fixes #21476

### CODE REVIEW

File: public/app/features/explore/TableContainer.tsx [hunk 1]
```diff
@@ -43,6 +43,13 @@ export class TableContainer extends PureComponent<TableContainerProps> {
...
     return (
       <Collapse label="Table" loading={loading} collapsible isOpen={showingTable} onToggle={this.onClickTableButton}>
+        {!tableResult?.length && (
+          <div className="logs-panel-meta">
+            <div className="logs-panel-meta__item">
+              <span className="logs-panel-meta__value">0 series returned</span>
+            </div>
+          </div>
+        )}
         {tableResult && <Table data={tableResult} width={tableWidth} height={height} onCellClick={onClickCell} />}
```
Comment (ivanahuckova) [inline public/app/features/explore/TableContainer.tsx]: Hi Lukasz, good job! One thing that I would change is to render either `0 series returned` or `Table`, not both if the tableReuslt.length is falsy. So maybe I would create variable `hasTableResult` and based on if it is falsy or truthy, I would render  `0 series returned` or `Table`. 
What do you think? See images as reference.
image

image

File: public/app/features/explore/TableContainer.tsx [hunk 2]
```diff
@@ -43,6 +43,13 @@ export class TableContainer extends PureComponent<TableContainerProps> {
 
     return (
       <Collapse label="Table" loading={loading} collapsible isOpen={showingTable} onToggle={this.onClickTableButton}>
+        {!tableResult?.length && (
+          <div className="logs-panel-meta">
+            <div className="logs-panel-meta__item">
+              <span className="logs-panel-meta__value">0 series returned</span>
+            </div>
+          </div>
```
Comment (torkelo) [inline public/app/features/explore/TableContainer.tsx]: this seems a bit hacky to use logs panel styles in the table

File: public/app/features/explore/TableContainer.tsx (hunk 1 above)
Comment (Estrax) [reply public/app/features/explore/TableContainer.tsx]: You're absolutely right. Fixed.

File: public/app/features/explore/TableContainer.tsx (hunk 2 above)
Comment (aocenas) [reply public/app/features/explore/TableContainer.tsx]: Yeah kinda breaks the BEM abstraction. What about creating a separate `MetaInfoText` component or something like that that would contain the styles. I think that is the way to share styles when using css-in-js solutions for styling.

Comment (Estrax) [reply public/app/features/explore/TableContainer.tsx]: @aocenas great idea with separating it to another component. On it.

**Review by ivanahuckova — COMMENTED** [review]
One more thing that I just thought of (sorry for not including it in initial feedback 🙈), it would be nice to have some unit tests testing these changes for `TableContainer` and for the new `MetaInfoText`.

**Review by ivanahuckova — APPROVED** [review]
Thanks for adding the tests! From my side - everything looks good! 🙂

**Review by aocenas — APPROVED** [review]
Looks great, let's merge this 👍

**Pull request merged**